EM.Sparql

```diff
@@ -0,0 +1,313 @@
+Assignment: SPARQL Queries for Ontology Evaluation
+
+This assignment outlines a series of SPARQL queries designed for evaluating and analyzing ontology structure and quality. 
+Each query addresses specific aspects of ontology evaluation, identifying potential issues or improvements necessary to enhance clarity and consistency.
+
+Detect Properties Without Domain or Range (level 8 - 0 points)
+Identifying Disjoint Classes with Common Instances (level 7 - 2 points)
+Detecting Unused Object Properties (level 7 - 2 points)
+Finding Classes with Conflicting Disjointness and Subclass Relationships (level 6 - 3 points)
+Identifying Classes with Different Labeling Standards (level 6 - 3 points)
+Identifying Potentially Overspecified Class Definitions (level 5 - 5 points)
+Detecting Equivalent Properties with Different Ranges  (level 5 - 5 points)
+Analyzing Class Hierarchies for Inconsistent Type Definitions (level 5 - 5 points)
+Detecting Potentially Misplaced Inverse Object Properties (level 5 - 5 points)
+Identifying Classes with No Properties (Level 4 - 10 points)
+Identifying Classes Without Labels (Level 4 - 10 points)
+Query to Detect Missing Annotations (Labels or Comments) Level 4 - 10 points
+Detecting Properties Shared Across Multiple CCO Classes Level 3 - 20 points
+Detecting Inconsistent Property Ranges Across Subclass Hierarchies Level 3 - 20 points
+
+1. Detect Classes Without Any Instances (Level 8 - 0 points)
+
+Title: Empty Class Detection
+Constraint Description: Identifies classes that have no instances associated with them and outputs the count of such classes.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT DISTINCT ?class (COUNT(?instance) AS ?instanceCount)
+WHERE {
+  ?class a rdfs:Class .
+  
+  # Filter out instances of the class to identify empty classes
+  OPTIONAL {
+    ?instance a ?class .
+  }
+} 
+GROUP BY ?class 
+HAVING (COUNT(?instance) = 0)
+
+Purpose: ​This query retrieves all classes that do not have any instances associated with them.​ 
+It uses FILTER NOT EXISTS to ensure that only classes without any defined instances are selected,helping to identify potentially unused or unnecessary classes in the ontology.
+
+2. Identifying Disjoint Classes with Common Instances (level 7 - 2 points)
+
+Title: Disjoint Class Violation
+Constraint Description: Detects instances that belong to classes declared as disjoint.
+Severity: Error
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+SELECT ?class1 ?class2 ?instance
+WHERE {
+  ?instance a ?class1 .
+  ?instance a ?class2 .
+  ?class1 owl:disjointWith ?class2 .
```

Comment:
Suppose you have a class A that is asserted in the OWL file to be owl:disjointWith itself. Suppose it has an instance a. This query would return: A, A, a. This is problematic and should be detected, but not for the motivation of this query. 
This suggests to me you should assert ?class1 =/ ?class2 and have a separate query that identifies as errors classes asserted to be disjoint with themselves.